mmengine/dataset/base_dataset.py

```diff
@@ -435,7 +439,7 @@ def load_data_list(self) -> List[dict]:
         """  # noqa: E501
         # `self.ann_file` denotes the absolute annotation file path if
         # `self.root=None` or relative path if `self.root=/path/to/data/`.
-        annotations = load(self.ann_file)
+        annotations = load(self.ann_file, backend_args=self.backend_args)
```

Comment:
Adding `backend_args` to the `load` function is a correct approach, but it may introduce **bc-breaking**. Assuming `self.ann_file` was originally a local path, providing `backend_args` could cause the `load` method to use **another file backend** (not local backend) to read the file (self.ann_file), but the file may not exist on that backend.

Suggesting avoiding this `bc-breaking`.

Reply:
Thank you for your comment! It's great to see that you're interested in understanding the potential bc-breaking issue. 
However, I'm not sure I understand how the bc-braking case may occur. Could you please give an example?

Thoughts:
In my world **when we pass** backend_args to the dataset we **want** this backend to be used and the annotation file is **the only file** that has to be read by the BaseDataset class by default. 
If one has annotation file on one backend, but other files (let's say images) on a different backend - this may be defined for each of the pipeline steps separately.
One possible solution to the `the file may not exist on that backend` issue might be to wrap FileNotFound exception inside the load function with a fallback to a local backend, but this option creates a biased decision towards local backend.

Thank you!

Reply:
Suppose the annotation file is an absolute path like `'/home/username/data/annotation.txt'`.
After adding `backend_args (such as AWSBackend)` for `load` function, it will try to load the file from AWS but it can not be found.

Reply:
Hi @tyomj , I opened a PR in [mmdet #10247](https://github.com/open-mmlab/mmdetection/pull/10247). In this case, the annotation can not be found in `petrel` backend.